broadleaf-thymeleaf3-presentation/src/main/java/org/broadleafcommerce/presentation/thymeleaf3/resolver/BroadleafThymeleaf3DatabaseResourceResolver.java

```diff
@@ -98,7 +99,8 @@ protected InputStream resolveResource() {
             }
             return null;
         } finally {
-            blcContextUtil.clearThinRequestContext();
+            BroadleafRequestContext brc = BroadleafRequestContext.getBroadleafRequestContext();
```

Comment:
How does this fix the issue?  What is the impact of not clearing the thinRequestContext?

Reply:
If we clear the context, the next step when we call BroadleafRequestContext.getCurrency() will create an empty context with no currency.
As for the impact of not clearing the thin RequestContext - I haven't found a problem, but it's best to double-check.

Reply:
@danielcolgrove This seems to fix everything but I'll do more testing. Do u have any questions? Is it ok to merge?

Reply:
I'm good if it tests out okay.